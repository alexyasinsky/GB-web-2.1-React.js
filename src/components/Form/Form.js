import { Button, Icon, TextField } from '@material-ui/core';

import { makeStyles } from '@material-ui/core/styles';
import {useRef, useEffect, useState} from "react";


export default function Form(props) {
	const handler = props.handler;
	const owner = props.user;

	const [message, setMessage] = useState();

	function createMessage(event) {
		event.preventDefault();
		const newMessage = {
			isOwner: true,
			user: owner,
			text: event.target.message.value,
			time: new Date().toLocaleTimeString()
		}
		setMessage(newMessage);
	}

	useEffect(() => {
		if (message) {
			handler(message);
		}
	}, [message])

	const inputRef = useRef(null);
	useEffect(() => {
		inputRef.current?.message.focus();
	});


	const useStyles = makeStyles((theme) => ({
		button: {
			margin: theme.spacing(1),
		},
		root: {
			'& .MuiTextField-root': {
				margin: theme.spacing(1),
				width: '25ch',
			},
		},
	}));

	const classes = useStyles();

	return (
		<form className='form' ref={inputRef} onSubmit={createMessage}>
			<TextField

				fullWidth={true}
				label="Message"
				name='message'
				multiline
				rows={4}
				variant="outlined"
			/>
			<Button variant="contained"
        color="primary"
        className={classes.button}
        endIcon={<Icon>send</Icon>}
        type='submit'
        >Отправить
			</Button>
		</form>
	)
}
