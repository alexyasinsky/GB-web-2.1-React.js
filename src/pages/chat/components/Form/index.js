import { Button, Icon, TextField, Grid } from '@material-ui/core';

import {useRef, useEffect, useState} from "react";


export default function Form(props) {
	const handler = props.handler;
	const owner = props.user;


	const [message, setMessage] = useState();

	const inputRef = useRef(null);
	useEffect(() => {
		inputRef.current?.message.focus();
	});

	function createMessage(event) {
		event.preventDefault();
		const newMessage = {
			isOwner: true,
			user: owner,
			text: event.target.message.value,
			time: new Date().toLocaleTimeString()
		};
		event.target.message.value = '';
		setMessage(newMessage);
	}

	useEffect(() => {
		if (message) {
			handler(message);
		}
	}, [message])


	return (
		<form ref={inputRef} onSubmit={createMessage}>
			<Grid container spacing={1} direction='column'>
				<Grid item>
					<TextField
						fullWidth={true}
						label="Message"
						name='message'
						multiline
						rows={4}
						variant="outlined"
					/>
				</Grid>
				<Grid item>
					<Button variant="contained"
					        color="primary"
					        endIcon={<Icon>send</Icon>}
					        type='submit'
					>Отправить
					</Button>
				</Grid>
			</Grid>
		</form>
	)
}
