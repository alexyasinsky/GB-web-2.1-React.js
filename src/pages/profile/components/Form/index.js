import { TextField } from '@material-ui/core';
import { Button } from '@material-ui/core';


export default function Form({hideModal, setUserName, userName, changeUserName}) {

  

  return (
    <div className='profile__form'>
      <div className='profile__inputBox'>
      <TextField
        id="name"
        label="Введите имя"
        variant="outlined"
        value={userName}
        onChange={setUserName}
      />
      <Button
        variant="contained"
        color='primary'
        onClick={changeUserName}
      >
        Изменить
      </Button>
      </div>
      
        {/* <div className='profile__inputBox'>
        <TextField
        id="avatar"
        label="Введите ссылку на фото для аватара"
        variant="outlined"
      />
      <Button
        variant="contained"
        color='primary'
      >
        Изменить
      </Button>
        </div> */}
      <Button
        variant="contained"
        color="secondary"
        onClick={hideModal}
      >
        Назад      
      </Button>
    </div>
    
  )
}