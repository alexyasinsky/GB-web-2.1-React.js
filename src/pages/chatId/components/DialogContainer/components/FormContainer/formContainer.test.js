import Form from "./index";
import {fireEvent, render, screen} from "@testing-library/react";


describe('Form Container tests', () => {
	test('соответствует ли компонент-контейнер снимку', () => {
		const component = render(<Form/>);
		expect(component).toMatchSnapshot();
	});
	// test('проверить запуск handler в методе createMessage компонента после нажатия кнопки', () => {
	// 	debugger
	// 	const handler = jest.fn();
	// 	render(
	// 		<Form handler={handler}/>
	// 	);
	// 	fireEvent.click(screen.getByTestId('submitButton'), {
	// 		target: {
	// 			message: {
	// 				value: 'message'
	// 			}
	// 		}
	// 	});
	// 	expect(handler).toBeCalled();
	// })
})