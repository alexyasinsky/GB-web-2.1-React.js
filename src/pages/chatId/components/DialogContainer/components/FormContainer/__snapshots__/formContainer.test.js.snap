// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Container tests соответствует ли компонент-контейнер снимку 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Message
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                  name="message"
                  rows="4"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Message
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="submitButton"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Отправить
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <span
                    aria-hidden="true"
                    class="material-icons MuiIcon-root"
                  >
                    send
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Message
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                name="message"
                rows="4"
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Message
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-testid="submitButton"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Отправить
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root"
                >
                  send
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
